render/gles2/renderer.c

```diff
@@ -331,6 +331,74 @@ static bool gles2_read_pixels(struct wlr_renderer *wlr_renderer,
 	return glGetError() == GL_NO_ERROR;
 }
 
+static bool gles2_blit_dmabuf(struct wlr_renderer *wlr_renderer,
+		struct wlr_dmabuf_attributes *dst_attr,
+		struct wlr_dmabuf_attributes *src_attr) {
+	assert(!wlr_renderer->rendering);
+
+	if (!gles2_procs.glEGLImageTargetRenderbufferStorageOES) {
+		return false;
+	}
+
+	bool r = false;
+
+	struct wlr_egl *egl = wlr_gles2_renderer_get_egl(wlr_renderer);
+	if (!wlr_egl_is_current(egl)) {
+		wlr_egl_make_current(egl, EGL_NO_SURFACE, NULL);
```

Comment:
The renderer is now expected to reset the EGL context when done, via `wlr_egl_unset_current`